smart-contracts/test/helpers/mainnet.js

```diff
@@ -28,9 +29,47 @@ const impersonate = async (address) => {
   })
   await addSomeETH(address) // give some ETH just in case
 }
+const stopImpersonate = async (address) => {
+  await network.provider.request({
+    method: 'hardhat_stopImpersonatingAccount',
+    params: [address],
+  })
+}
+
+const toBytes32 = (bn) => {
+  return ethers.utils.hexlify(ethers.utils.zeroPad(bn.toHexString(), 32))
+}
+
+const addUDT = async (recipientAddress, amount = 1000) => {
+  const { chainId } = await ethers.provider.getNetwork()
+
+  // UDT contract
+  const { address: udtAddress } = getDeployment(
+    chainId,
+    'UnlockDiscountTokenV2'
+  )
+  const udtAmount = ethers.utils.parseEther(`${amount}`)
+
+  // NB: slot has been found by using slot20 - see https://kndrck.co/posts/local_erc20_bal_mani_w_hh/
+  // Get storage slot index
+  const index = ethers.utils.solidityKeccak256(
+    ['uint256', 'uint256'],
+    [recipientAddress, 51] // key, slot
+  )
+
+  // Manipulate local balance (needs to be bytes32 string)
+  await network.provider.request({
+    method: 'hardhat_setStorageAt',
+    params: [udtAddress, index.toString(), toBytes32(udtAmount).toString()],
+  })
```

Comment:
This helper to give UDT works but it is a bit of a dirty hack that modify directly the contract storage. Maybe it is better to use a more proper way, like impersonate Unlock, then mint and transfer tokens.

Reply:
For instance it will make `totalSupply` wrong